src/main/java/org/mifos/connector/channel/zeebe/ZeebeWorkers.java

```diff
@@ -251,4 +257,55 @@ private void zeebeConsistencyWorker() {
                 .maxJobsActive(workerMaxJobs)
                 .open();
     }
+<<<<<<< Updated upstream
+=======
+
+    //validate transactional data
+    private void validateTransactionalData(){
+        zeebeClient.newWorker()
+                .jobType("validate-transactional-data")
+                .handler((client, job) -> {
+                    logger.info("Job '{}' started from process '{}' with key {}", job.getType(), job.getBpmnProcessId(), job.getKey());
+                    Map<String, Object> variables = job.getVariablesAsMap();
+                    ObjectMapper mapper = new ObjectMapper();
+                    TransactionChannelRequestDTO channelRequestDTO = mapper.readValue(variables.get(CHANNEL_REQUEST).toString(),
+                            TransactionChannelRequestDTO.class);
+                    boolean valid = verifyTxn(channelRequestDTO);
+                    if(!valid){
+                        variables.put(TRANSACTION_VALID, false);
+                        variables.put(ERROR_INFORMATION, "Custom Error: Transaction Invalid");
+                        variables.put(ERROR_DESCRIPTION,"Transaction data was wrong or incomplete");
+                    } else{
+                        variables.put(TRANSACTION_VALID, true);
+                    }
+                    client.newCompleteCommand(job.getKey())
+                            .variables(variables)
+                            .send()
+                            .join();
+                })
+                .name("validate-transactional-data")
+                .maxJobsActive(workerMaxJobs)
+                .open();
+
+
+    }
+
+    private boolean verifyTxn(TransactionChannelRequestDTO channelRequestDTO) {
```

Comment:
can we rename this method to validateTxn

Reply:
okay